src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -988,6 +990,7 @@ JavaThread* JavaThread::active() {
 }
 
 bool JavaThread::is_lock_owned(address adr) const {
+  assert(!LockingMode != 2, "should not be called with new lightweight locking");
```

Comment:
Looks like there is a typo here. I think this should be:
```
assert(LockingMode != 2, "should not be called with new lightweight locking");
```

Reply:
> Looks like there is a typo here. I think this should be:
> 
> ```
> assert(LockingMode != 2, "should not be called with new lightweight locking");
> ```

Thanks for catching this! I pushed a fix.
You might want to try the riscv port again, I have made some changes that might have broken it. Also, you might want to compare it again with aarch64 (or x86_64), I have done a bunch of changes and improvements that you might want to adopt.

Thanks!
Roman